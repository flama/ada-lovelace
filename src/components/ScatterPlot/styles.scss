.scatter-plot {
  flex-basis: auto;
  display: flex;
  flex-direction: column;
  height: 120vh;
}
