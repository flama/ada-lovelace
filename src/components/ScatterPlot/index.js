import React, { Component } from 'react'
import ScatterRow from '../ScatterRow'

import './styles.scss'

const domain = {
  x: [-1800, 2030]
}

class ScatterPlot extends Component {

  render() {

    //TODO: change to actual data
    let topics = Object.keys(this.props.dataList).map(category => {
      return (<ScatterRow
        data={ this.props.dataList[category] }
        domain={ domain }
        title={ category }
        key={ category }
      />)
    })

    return (
      <div className="scatter-plot">
        { topics }
      </div>
    )
  }
}

export default ScatterPlot
