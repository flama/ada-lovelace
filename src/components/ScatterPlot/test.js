import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import ScatterPlot from '../ScatterPlot'

test("the truth", () => {
  expect(true).toBeTruthy()
})
