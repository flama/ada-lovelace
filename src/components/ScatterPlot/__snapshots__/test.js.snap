// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScatterPlot renders correctly 1`] = `
<div
  className="scatter-plot"
>
  <div
    className="topics"
  />
  <DetailsBubble
    close={[Function]}
  />
  <div
    className="scatter-row"
  >
    <div
      className="title"
    />
    <GraphTimeline
      className="graph"
      domain={
        Object {
          "x": Array [
            1800,
            2017,
          ],
        }
      }
    />
  </div>
</div>
`;
